.github/workflows/DetectDocGeneratorChanges.yml

```diff
@@ -0,0 +1,52 @@
+name: Detect Doc Generator Changes
+
+on: [pull_request]
+
+permissions:
+  id-token: write
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        role-to-assume: ${{ secrets.CI_AWS_ROLE_ARN }}
+        role-duration-seconds: 7200
+        aws-region: us-west-2
```

Comment:
Does this job need AWS credentials? My impression was that it shouldn't to read the recipe files and then transform to markdown.